ls1intum/Artemis
# Feature/programming exercise/complaints

### Checklist

### Motivation and Context


We need to enable complaints for programming exercises (for manual results). For that, it would be best to use the complaint component on the exercise details page.

### Description

I created a new complaint interactions component (feedback + complaints) from the text exercises implementation (refactored the related code). This component is now also used on the course exercise details page if the exercise is a programming exercise. 

Disclaimer: There is a bug, where the `currentResult` in the `course-exercise-details.component` changes after a couple of cycles from the latest (correct in the beginning) to the oldest. Even after debugging and searching for a long time, I could not find the reason for this. If anyone has some insight on this, it would be much appreciated. For now, I just sort the array again before fetching the latest result.

### Steps for Testing


1. Log in to Artemis
2. Create a programming exercise with manual reviews
3. Participate
4. You should not be able to submit a complaint
5. Submit a manual review (e.g. as an instructor action)
6. You should be able to complain or request more feedback on the details page (see screencast)

### Screenshots


![complaint_programming](https://user-images.githubusercontent.com/27979674/68419854-540e3f80-019b-11ea-96ca-7672b8ea66ff.gif)
